### ISSUE TITLE
[GDnative] Disable lto for gdnative wrapper static lib

### ISSUE THREAD

**Issue by touilleMan** [godotengine/godot#14976]
Given the very nature of the gdnative wrapper lib (i.e. being compiled by 3rd party to expose) it can be compiled with a wide range of compilers. However lto options seems to go in the way:

With gcc 5.4 the compilation errored with a `lto1: fatal error: bytecode stream generated with LTO version 6.0 instead of the expected 4.1`. This is not cool because most recent version of gcc are not widely available on distributions (especially on CI platforms such as travis).

With clang the compilation succeed but symbols exposed by the wrapper lib are stripped out:
```
$ nm libpythonscript.so | grep gdnative_wrapper_
nm: libpythonscript.so: plugin needed to handle lto object
                 U _gdnative_wrapper_api_struct
                 U _gdnative_wrapper_arvr_api_struct
                 U _gdnative_wrapper_nativescript_api_struct
                 U _gdnative_wrapper_pluginscript_api_struct
```

On the other hand I'm not sure lto brings any benefit here (given this static library is small and all symbols should be used anyway...). I'm far from being an expert in this area so I might be missing something here.

**Comment by karroffel** [comment]
I'm not familiar with LTO, but since the static library is just a "byproduct" in the build I don't see why it would be affected of LTO? Like I said, I'm no expert. If we can just disable it for that library it might help?

cc @hpvb

**Comment by touilleMan** [comment]
> I don't see why it would be affected of LTO

LTO is a flag passed to the compiler, given it is define by `platform/<platform>/detect.py` it is part of the main scons environment which is cloned to create the gdnative wrapper scons environment.

From what I saw, GCC supports the `-fno-lto` option which could be passed in the clone scons env to shadow the main configuration.

On windows we can do similar thing by passing `/GL-` (see https://docs.microsoft.com/en-us/cpp/build/reference/gl-whole-program-optimization) and `/LTCG:OFF` (see https://docs.microsoft.com/en-us/cpp/build/reference/ltcg-link-time-code-generation). The thing is I'm not sure if it useful given I was able to compile the gdnative wrapper dll (however I didn't run the resulting binary given I don't have a windows computer until next week, if somebody is interested to try the builds are here https://github.com/touilleMan/godot-python/releases/tag/v0.10.0-rc1)
